src/qt/calctrl.cpp

```diff
@@ -168,15 +180,22 @@ bool wxCalendarCtrl::SetDateRange(const wxDateTime& lowerdate,
 bool wxCalendarCtrl::GetDateRange(wxDateTime *lowerdate,
                                   wxDateTime *upperdate) const
 {
+    bool status = true;
+
     if ( !m_qtCalendar )
-        return false;
+        status = false;
 
-    if (lowerdate)
+    if ( lowerdate && IsQDateValid( m_qtCalendar->minimumDate() ) )
```

Comment:
How will this not crash when `m_qtCalendar` is null but you no longer `return` in that case, 2 lines above?

Reply:
A very good question, I will do some testing around this and get back to you!

Reply:
In short... it will crash! Do you think the best idea would be to remove the boolean `status` flag altogether and simply have multiple return statements or to keep the `status` flag and change line 188 to: `return false`?

Reply:
I don't know the code, but the best thing is almost always to start with the minimal changes that are needed to achieve your goal (see the removal of an empty line that could probably have been avoided too).

Reply:
I have made a further change which should hopefully address these issues. Many thanks for your input @catalinr